assessments/projects/TiKV/self-assessment.md

```diff
@@ -0,0 +1,363 @@
+
+<img src="./src/imgs/tikv_wholepic.png">
+
+### Actors
+
+* **Clients**
+* **TiKV Instance**
+  - Data Regions (within TiKV Nodes)
+* **Placement Driver**
+    - TimeStamp Oracle (within Placement Driver)
+
+### Actions
+
+* **Clients**
+  - TiKV provides two types of APIs for developers: the Raw Key-Value API and the Transactional Key-Value API
+* **TiKV Instance**
+  - Each instance acts isolated from every other node 
+  - Data is distributed across TiKV instances via the Raft Consensus Algorithm
+  - Data Regions (within TiKV Nodes)
+    - This is the basic unit of key-value data movement
+    - Each Region is replicated to multiple Nodes, and multiple replicas form a Raft group
+* **Placement Driver**
+    - The cluster manager of TiKV periodically checks replication constraints to balance load and data automatically across nodes and regions in a process called auto-sharding
+    - TimeStamp Oracle (within Placement Driver)
+      - Plays a significant role in the Percolator transaction model, where every transaction requires contacting the Oracle at least twice and proper scalability
+
+### Goals
+
+* TiKV clients let you connect to a TiKV cluster and use raw (simple get/put) API or transaction (with transactional consistency guarantees) API to access and update your data. TiKV clients interact with PD and TiKV through gRPC
+* Scalability and consistency due to data being shared and replicated across multiple nodes (up to petabytes)
+* Ensure data confidentiality and integrity in transit and at rest, using TLS
+* Guarantee ACID properties for transactions to prevent data corruption
+* Externally consistent distributed transactions to operate over multiple Key-Value Pairs
+* Develop and maintain sophisticated access control systems to manage permissions effectively for different users and services interacting with TiKV
+* Ensure compatibility and integration with widely used cloud-native security tools and platforms for seamless deployment in various environments 
+* TiKV adopts MVCC (Multiversion Concurrency Control) which offers an extra layer of security in data management by maintaining multiple versions of data records. This approach helps to prevent conflicts and inconsistencies in a distributed environment where concurrent transactions occur 
+
+
+### Non-goals
+* Not supposed to be a relational database (No SQL)
+* Does not specifically implement features which compromise performance without clear security benefits
+* Perfect security is not a goal of TiKV; the aim should be to mitigate risk to acceptable levels as perfect security is unattainable
+* Does not address end-user security, such as password manager or client side security, which are outside the database’s control
+* Does not attempt to solve external network security issues that are beyond the scope of TiKV
+
+## Self-assessment use
+
+This self-assessment is created by the TiKV team to perform an internal analysis of the
```

Comment:
You mention elsewhere in this document that the TiKV team was not available to participate in this self assessment, so this sentence appears inaccurate. Consider adjusting this sentence and using the topmost paragraph (below the first header) to list the names of all contributors to this document.

Reply:
We added our names below the Overview Header and also updated the line you mentioned above (line 148). We wrote that line originally assuming that the TiKV project maintainers would be able to get back to us. Thank you for bringing it to our attention. If our changes do not reflect what you meant in your suggestion, please let us know!